enterprise/backend/src/metabase_enterprise/content_translation/api/dictionary.clj

```diff
@@ -29,19 +66,61 @@
     (str/blank? msgstr)
     (re-matches #"^[,;\s]*$" msgstr))))
 
+(defn- validate-row
+  "Validate a single row of CSV data, recording errors and updating the set of seen translation keys."
+  [row-index row seen-keys]
+  (let [errors (transient [])
```

Comment:
I'm curious if the use of `transient`s here and elsewhere is because of performance considerations that we've identified and profiled or is this pre-emptive?

Generally see transients as a good option to speed up hot spots but generally lead to less idiomatic clojure code, and here it seems like the number of changes we're making to the `errors` transient is bounded and relatively small, so I think I'd lean towards something more idiomatic.

`(remove nil? [(collect-duplication-error ...) (collect-locale-error ...) (collect-string-length-error ...) ...])` for example would just collect the non-nil error results (I guess you should verify that we're actually returning `nil` in the non-error case, instead of `false` or something)

Reply:
Thanks for the feedback! I've taken out the transients.